src/java.base/share/classes/java/lang/Object.java

```diff
@@ -127,15 +127,26 @@ public Object() {}
      * <li>For any non-null reference value {@code x},
      *     {@code x.equals(null)} should return {@code false}.
      * </ul>
+     *
      * <p>
+     * An equivalence relation partitions the elements it operates on
+     * into <i>equivalence classes</i>; all the members of an
+     * equivalence class are equal to each other. The equal objects
+     * are substitutable for each other, at least for some purposes.
+     *
+     * @implSpec
      * The {@code equals} method for class {@code Object} implements
      * the most discriminating possible equivalence relation on objects;
      * that is, for any non-null reference values {@code x} and
      * {@code y}, this method returns {@code true} if and only
      * if {@code x} and {@code y} refer to the same object
      * ({@code x == y} has the value {@code true}).
-     * <p>
-     * Note that it is generally necessary to override the {@code hashCode}
+     *
+     * In other words, under the reference equality equivalence
+     * relation, each equivalence class only has a single element.
+     *
+     * @apiNote
+     * Note that it is generally necessary to override the {@link hashCode hashCode}
```

Comment:
The `@apiNote` introduces the paragraph with a subhead "API Note:" so it's a bit redundant to say "Note that...." Maybe just start with "It is generally necessary...."

Reply:
Agree.